src/util.cc

```
@@ -221,6 +221,39 @@ int WriteFileSync(v8::Isolate* isolate,
   return WriteFileSync(path, buf);
 }
 
+int ReadFileSync(std::string* result, const char* path) {
+  uv_fs_t req;
+  uv_file file = uv_fs_open(nullptr, &req, path, O_RDONLY, 0, nullptr);
+  if (req.result < 0) {
+    return req.result;
+  }
+  uv_fs_req_cleanup(&req);
+
+  auto defer_close = OnScopeLeave([file]() {
+    uv_fs_t close_req;
+    CHECK_EQ(0, uv_fs_close(nullptr, &close_req, file, nullptr));
+    uv_fs_req_cleanup(&close_req);
+  });
+
+  *result = std::string("");
+  char buffer[4096];
+  uv_buf_t buf = uv_buf_init(buffer, sizeof(buffer));
+
+  while (true) {
+    const int r =
+        uv_fs_read(nullptr, &req, file, &buf, 1, result->length(), nullptr);
+    if (req.result < 0) {
+      return req.result;
```

It seems like uv_fs_req_cleanup is not called on this fail fast path.